File: kinotic-js/workspace/packages/management-api/src/api/model/telemetry/MetricQuery.ts
/**
 * Parameters for a metric query: a PromQL expression evaluated over one organization's metrics
 * at a fixed step across a time range.
 */
export interface MetricQuery {

    /**
     * Organization whose metrics to query. Null names the platform's own, which only a system
     * participant may read.
     */
    organizationId: string | null

    /** The PromQL expression to evaluate, e.g. `sum by (service) (rate(traces_spanmetrics_calls_total[5m]))`. */
    query: string

    /** Start of the time range, epoch milliseconds (inclusive). */
    start: number

    /** End of the time range, epoch milliseconds (inclusive). */
    end: number

    /** Resolution of the result, in seconds between evaluated points. */
    step: number
}

File: kinotic-management-api/src/main/java/org/kinotic/management/api/model/MetricQuery.java
package org.kinotic.management.api.model;

import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.Accessors;

/**
 * Parameters for a metric query: a PromQL expression evaluated over one organization's metrics
 * at a fixed step across a time range.
 */
@Getter
@Setter
@Accessors(chain = true)
@NoArgsConstructor
public class MetricQuery {

    /**
     * Organization whose metrics to query. Null names the platform's own, which only a system
     * participant may read.
     */
    private String organizationId;

    /**
     * The PromQL expression to evaluate, e.g.
     * {@code sum by (service) (rate(traces_spanmetrics_calls_total[5m]))}.
     */
    private String query;

    /**
     * Start of the time range, epoch milliseconds (inclusive).
     */
    private long start;

    /**
     * End of the time range, epoch milliseconds (inclusive).
     */
    private long end;

    /**
     * Resolution of the result, in seconds between evaluated points.
     */
    private long step;
}
